pkg/gadget-manifest/instance-spec.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gadgetmanifest
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	Version          = 1
+	KindInstanceSpec = "instance-spec"
+)
+
+type InstanceSpec struct {
+	Version     int               `json:"version" yaml:"version"`
+	Kind        string            `json:"kind" yaml:"kind"`
+	Image       string            `json:"image" yaml:"image"`
+	ID          string            `json:"id" yaml:"id"`
+	Name        string            `json:"name" yaml:"name"`
+	Tags        []string          `json:"tags" yaml:"tags"`
+	ParamValues map[string]string `json:"paramValues" yaml:"paramValues"`
```

Comment:
Just to confirm: `map[string]string `  means we don't support:
```
paramValues:
  operator:
    foo: baz
    bar:  baz
```

Reply:
Yeah, we don't. Do you think it's not okay to not support that?

Reply:
We support it in [configuration file](https://www.inspektor-gadget.io/docs/latest/reference/configuration#configuration-file) so was wondering if we should keep it same here but I don't have too strong opinion on it!

Reply:
I think supporting only the "dot" notation is better. It's the same notation used by the Golang API and very similar to the CLI (in the CLI we can use shorter names).